apis/network/v1alpha2/subnet_referencers.go

```diff
@@ -0,0 +1,66 @@
+/*
+Copyright 2019 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha2
+
+import (
+	"context"
+
+	"github.com/crossplaneio/crossplane-runtime/pkg/resource"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	runtimev1alpha1 "github.com/crossplaneio/crossplane-runtime/apis/core/v1alpha1"
+
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// SubnetIDReferencer is used to get the name from another Subnet
+type SubnetIDReferencer struct {
+	corev1.LocalObjectReference `json:",inline"`
+}
+
+// GetStatus implements GetStatus method of AttributeReferencer interface
+func (v *SubnetIDReferencer) GetStatus(ctx context.Context, _ resource.CanReference, reader client.Reader) ([]resource.ReferenceStatus, error) {
+	refObj := Subnet{}
+	nn := types.NamespacedName{Name: v.Name}
+	if err := reader.Get(ctx, nn, &refObj); err != nil {
+		if kerrors.IsNotFound(err) {
+			return []resource.ReferenceStatus{{Name: v.Name, Status: resource.ReferenceNotFound}}, nil
+		}
+
+		return nil, err
+	}
+
+	if !resource.IsConditionTrue(refObj.GetCondition(runtimev1alpha1.TypeReady)) {
+		return []resource.ReferenceStatus{{Name: v.Name, Status: resource.ReferenceNotReady}}, nil
+	}
+
+	return []resource.ReferenceStatus{{Name: v.Name, Status: resource.ReferenceReady}}, nil
+}
+
+// Build retrieves and builds the NetworkName
+func (v *SubnetIDReferencer) Build(ctx context.Context, _ resource.CanReference, reader client.Reader) (string, error) {
+	refObj := Subnet{}
+	nn := types.NamespacedName{Name: v.Name}
+	if err := reader.Get(ctx, nn, &refObj); err != nil {
+		return "", err
+	}
+
+	return refObj.Status.ID, nil
```

Comment:
Does this field return a value like `/subscriptions/${SUBSCRIPTION_ID}/resourceGroups/demo-rg/providers/Microsoft.Network/virtualNetworks/demo-vnet/subnets/demo-subnet`?

Reply:
it does :)